subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyHandler.java

```diff
@@ -461,7 +460,7 @@
      * @since 5.3
      */
     @Incubating
-    <T extends TransformAction> void registerTransformAction(Class<T> actionType, Action<? super TransformSpec> registrationAction);
+    <T> void registerTransformAction(Class<? extends TransformAction<T>> actionType, Action<? super ParameterizedTransformSpec<T>> registrationAction);
```

Comment:
If we want to go down the route shown in this document, this method will be renamed to `registerTransform` and replace the current `registerTransform`.